MIPS/mip-x.md

```diff
@@ -0,0 +1,246 @@
+3. **Feed liquidations back into the protocol** Luquidations should go back into the mStable protocol to help further improve APY through Swap Fees. 
+
+Some thought was given to implementing more exotic Yield Farming strategies to
+increase the APY further. Depositing to CREAM was considered as was depositing
+to Balancer Pools or to Sake Pools. This was felt to be too risky and to
+require active Fund Management to ensure good returns. 
+
+As such an automated, predictable return is favoured by selling `Token A` for
+`Token B` on the open market. 
+ 
+Related work includes yearn.finance selling $CRV tokens on the open market. 
+
+### Technical Specification
+<!--The technical specification should outline the public API of the changes proposed. That is, changes to any of the interfaces mStable currently exposes or the creations of new ones.-->
+
+A `Liquidator.sol` contract will be created that will sell `Token A` for `Token
+B` on a DEX. Initially Balancer will be supported but 1inch, UniSwap and
+other integrations may be added. In the short term the contract will be
+upgradable with a view to it becoming immutable. 
+
+Protocol contracts that hold reward tokens will give the Liquidator infinite
+approval to spend the rewards tokens. The Liquidator will then begin selling
+`Token A` and will return `Token B` to originating contract.
+
+An initial use case is to sell the $COMP available on the
+`CompoundIntegration.sol` contract for $cDAI. The `CompoundIntegrator.sol`
+contract will grant the `Liquidator.sol` contract infinite approval on the
+$COMP that it holds. The Liquidator will begin selling the $COMP at a drip rate
+and will return $cDAI to the `CompoundIntegrator.sol` contract. 
+
+To prevent people gaming Liquidations and depositing large amounts around a
+liquidation to benefit from the returns liquidations will be smoothed over time
+and separated into two steps.
+
+The first is for the Liquidator to sell a proportion of the amount it holds
+when the `triggerLiquidation` function is called. The `triggerLiquidation` is a
+public function and may be called by anyone, but may only be called once in a
+24 hour period. Based on the spot price for the asset the function will compute
+a value between $1000 and $3000 and trigger a sale for this amount. After the
+token has been swapped the token remains on the Liquidator contract.
+
+The second step is for the Integration contract to call `collect()` on the
+liquidator. This function will transfer a proportion of the tokens from the
+Liquidator contract to the Integration contract. At this point the token is on
+the Integration contract and earning Supply APY.
+
+The following psueudo code is a proposal for how `collect()` function should
+distribute bought tokens to the Integration contract when called. 
+
+```
+contractBal = buyAsset.balanceOf(address(this)) 
```

Comment:
I would add some randomness in to the collection schedule.

e.g. if `timeSinceLastCollection` > `1 hour * random multiplier`  (i.e. has it been between 1 & say, 3 hours since last collection?)